### ISSUE TITLE
Misclick in hand menu

### ISSUE THREAD

**Issue by paulmelis** [StereoKit/StereoKit#425]
## Description

See https://user-images.githubusercontent.com/1408287/177714366-2a0f9b96-4ac2-4c40-a23f-6b51a355f3e2.mp4. With a simple (linear) hand menu of `UI.Radio`'s I get the wrong button being activated, even though the circle indicator appears over the correct button. However, the correct button does lose its highlight when moving the finger closer, so something weird is definitely going on. With some more testing it appears that it's only ever the button *below* the one being attempted to click which gets misclicked.
Perhaps a button bounds-checking issue? 

## Platform / Environment

Windows, Varjo XR-3, observed with SK 0.3.6 and 0.3.7-preview1

## Logs or exception details

No errors or exceptions in the logs

**Comment by paulmelis** [comment]
For completeness, here's how the menu it set up:

```
            if (!HandPalmFacingHead(Handed.Left))
                return;

            Vec2 size = new Vec2(4, 16);

            // Position the menu relative to the side of the hand
            Hand hand = Input.Hand(Handed.Left);
            Vec3 at = hand[FingerId.Little, JointId.KnuckleMajor].position;
            Vec3 down = hand[FingerId.Little, JointId.Root].position;
            Vec3 across = hand[FingerId.Index, JointId.KnuckleMajor].position;

            Pose menuPose = new Pose(
                at,
                Quat.LookAt(at, across, at - down) * Quat.FromAngles(0, 90, 0));
            menuPose.position += menuPose.Right * (-3 - size.x) * U.cm;
            menuPose.position += menuPose.Up * (size.y / 2 + 5f) * U.cm;

            UI.WindowBegin("ConfigMenu", ref menuPose, size * U.cm, UIWin.Empty);
            UI.Label("View:");
            List<ViewConfiguration> configs = scenes[current_scene_index].configurations;            
            for (int i = 0; i < configs.Count; i++)
            {
                if (UI.Radio(configs[i].name, i == current_configuration_index))
                    SetConfiguration(i);
            }
            UI.WindowEnd();
```

**Comment by paulmelis** [comment]
By the way, what exactly is being tested for button volume intersection, the finger tip, or the full hand mesh? As I'm doing this on an XR-3 with the ultimate fat finger meshes (still not fixed in UltraLeap's OpenXR layer).

**Comment by maluoi** [comment]
Actually, yeah, fat fingers are a good hypothesis! SK's UI volumes are a little expanded to help with hit detection, and its possible the oversized fingers may be affecting those. `UI.ShowVolumes` will show the collision volumes, and might help illustrate what's happening a bit better. If this is happening, then the UI element "downstream" in the code will be the one that snags the focused flag. But I'd also expect you'd run into this sort of issue a lot more often if so?

If that's the case, there may be some UI element priority work that I can do to help cases where multiple buttons overlap. I may also no longer need as much volume expansion anymore due to improvements in hit detection code too! I'll look into these.

If you want to verify the hashes yourself, there's a `UI.StackHash` function :) Just make sure to use it in the same part of the UI hierarchy, as it is sensitive to what hashes the windows/handles have pushed onto the stack. I did check it with the same strings as were in your video!

**Comment by paulmelis** [comment]
So I finally found time after work and holidays to test with UI volumes on. With my simple hand menu it seems the interaction volumes overlap between neighboring buttons, which together with the fat fingers might account for the misclick behaviour?

https://user-images.githubusercontent.com/1408287/187927450-a0a967ca-97b8-40a4-9cb7-630f1715f353.mp4

**Comment by maluoi** [comment]
Yeah, those are _enormous_ fingers! In the `feature/ui_interactors` branch, I _should_ have this fixed, I did switch to a priority system that accounts for distance from the button volumes. I haven't tested with fingers like these yet, but that should be a good stress test for 'em :D

I still have to finish up that code though, it's not quite so usable yet. It's a lot more than just the priority system in there!

**Comment by paulmelis** [comment]
Related to #337, have the ui_interactors changes you mentioned above been merged already?

**Comment by maluoi** [comment]
I unfortunately put that one on the backburner :( I'd like to get back to it, but I'm not too certain when I'll be revisiting it.

**Comment by paulmelis** [comment]
Using `SetHandJointScale` from #669 to something like 0.6 in my case definitely helps in avoiding misclicks (and it also makes the hands look much better)! But I will need to do a field test with actual users to see if theory matches practice :)
